### PR TITLE
fix flag link in resources

### PR DESCRIPTION
**Pull request by davidbgomes**
[Contries flags from resources landing Page should be linked to each country](https://app.clickup.com/t/865cjfu6e)

- Fix filters & flag link

### CODE REVIEW

File: src/webapp/components/resources/resources-filters/ResourcesFilters.tsx [hunk 1]
```diff
@@ -20,13 +20,14 @@ export const ResourcesFilters: React.FC<ResourcesFiltersProps> = React.memo(
 
         const onCategoryChange = useCallback(
             (value: string | undefined) => {
+                filters.country.onSelectedChange([]);
```
Comment (xurxodev) [inline src/webapp/components/resources/resources-filters/ResourcesFilters.tsx:23]: This is a presentation logic and should be in the hooks

File: src/webapp/pages/resources/by-type/useResourcesByType.ts [hunk 2]
```diff
@@ -15,17 +15,15 @@ export function useResourcesByType(params: {
 
     const [loading, setLoading] = useState<boolean>(false);
     const [items, setItems] = useState<ResourceState[] | undefined>(undefined);
+    const [selectedCountry, setSelectedCountry] = useState<string[] | undefined>([""]);
```
Comment (xurxodev) [inline src/webapp/pages/resources/by-type/useResourcesByType.ts:18]: It's confusing and dangerous to have two states fo the country selected: selectedCountry here and filter.country.selected
I think should found the solution to have only the filter.country.selected

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @davidbgomes 

I have found some issues to review, I've added comments in the code

**Comment by davidbgomes** [comment]
Thank you for the review @xurxodev. I wasn't able to make it work without those points that you stated above.

- On the first one I moved from the hook so the selected country appears on the dropdown (otherwise it would re-render and never show it)

- On the second point it would for some reason reset the `filters.country.selected` on some occasions, I would guess some race condition but using a separate state worked.

Maybe we can look at it together?

**Comment by xurxodev** [comment]
@davidbgomes Finally I fixed the problems in this PR:
I have realized two fixes:
* Sync country param with countries selection in dropdown in two directions
* Fix overlapping request: this was the problem because sometimes the filter doesn't works. I use cancel of futures top avoid overlapping

Please apply the solution in this PR in https://github.com/EyeSeeTea/uhcpw/pull/73

**Review by xurxodev — APPROVED** [review]
thanks @davidbgomes

**Comment by davidbgomes** [comment]
Thanks for the explanation @xurxodev , nice to know for the future (no pun intended). I also tested and works great. Already applied the solution for the theme PR.

**Pull request merged**